src/app/modules/myinfo/myinfo.controller.ts

```diff
@@ -194,8 +195,10 @@ export const loginToMyInfo: ControllerHandler<
     })
     return res.sendStatus(StatusCodes.BAD_REQUEST)
   }
-  const { formId, cookieDuration } = parseStateResult.value
-  const redirectDestination = `/${formId}`
+  const { formId, cookieDuration, encodedQuery } = parseStateResult.value
+  const redirectDestination = encodedQuery
+    ? `/${formId}?${Buffer.from(encodedQuery, 'base64').toString('utf8')}`
```

Comment:
may need a try-catch here if the Buffer is wrongly encoded

Reply:
good point, changed